pkg/safecli/safecli.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safecli
+
+import "strings"
+
+const (
+	redactedValue     = "<****>"
+	keyValueDelimiter = "="
+)
+
+// Redactor defines an interface for handling sensitive value in the way
+// that it can be represented both in a plain and redacted text form.
+type Redactor interface {
+	PlainString() string
+	String() string
+}
+
+// redactor defines a function that converts value to Redactor.
+type redactor func(value string) Redactor
```

Comment:
I'd prefer to have more clear name for this function. Let's say `createRedactor` or something like that.